src/App.tsx

```diff
@@ -1,19 +1,64 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Error } from './components/Error';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortFilter, setSortFilter] = useState('all');
+  const [visibelTodos, setVisibelTodos] = useState<Todo[]>([...todos]);
+
+  const [isError, setError] = useState(false);
+  const [messageError, setMessageError] = useState('');
+
+
+  const deleteCompletedTodos = () => (
+    setTodos(state => (
+      state.filter(todo => !todo.completed) || []))
+  );
```

Comment:
```suggestion
  const deleteCompletedTodos = () => (
    setTodos(todos => (
      todos.filter(todo => !todo.completed)))
  );
```
A variable name should describe what is stored in it.